web/src/App.tsx

```diff
@@ -228,4 +227,81 @@ function App () {
   )
 }
 
+function DropPlaceholder() {
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen w-full text-gray-200">
+      <h2 className="text-xl font-semibold mb-2">Drop Here!</h2>
+    </div>
+  )
+}
+
+function FilePicker({ onFilePicked }: { onFilePicked: (file: File | undefined) => void }) {
+  const handleFileSelected = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onFilePicked(e.target.files?.[0]);
+  };
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen w-full text-gray-200">
+      <h2 className="text-xl font-semibold mb-2">Drop Dep-Tree JSON Graph File Here</h2>
+      <p className="text-gray-400 mb-4">or</p>
+      <input
+        type="file"
+        accept=".json"
+        onChange={handleFileSelected}
+        className="bg-gray-800 px-4 py-2 rounded cursor-pointer hover:bg-gray-700"
+      />
+    </div>
+  )
+}
+
+async function readFileText(file: File): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.onload = () => resolve(reader.result as string)
+    reader.onerror = reject
+    reader.readAsText(file)
+  })
+}
+
+function App() {
+  const [graphData, setGraphData] = useState<Graph>(Data.__INLINE_DATA)
+  const [isDragging, setIsDragging] = useState(false)
+  const [forceRemountKey, setForceRemountKey] = useState(0)
+
+  async function handleFilePicked(file: File | undefined) {
+    if (!file) return
+    try {
+      const text = await readFileText(file)
+      const data = JSON.parse(text)
+      setGraphData(data)
+      setForceRemountKey((x) => x + 1)
+    } catch (error) {
+      console.error('Error reading file:', error)
+    }
+  }
+
+  function handleDrop(e: React.DragEvent<HTMLDivElement>) {
+    e.preventDefault()
+    setIsDragging(false)
+    handleFilePicked(e.dataTransfer.files[0])
```

Comment:
Rather than implicitly ignoring a promise returned by this function, I would ignore it explicitly for the shake of readability:
```suggestion
    void handleFilePicked(e.dataTransfer.files[0])
```